### ISSUE TITLE
RESTART: /Users/myusername/Downloads/instabot-master 2/instabot.py

### ISSUE THREAD

**Issue by alvvvvvin** [LevPasha/instabot.py#1]
Hi. Thanks for posting this. Trying to test if it works. I've ran the script in IDLE. After finding a resolve for properly installing "REQUESTS" & "LXML", I'm stuck w/:

====== RESTART: /Users/myusername/Downloads/instabot-master 2/instabot.py ======

Any help would be great. Thanks!

**Comment by LevPasha** [comment]
Hi! What OS do you use? I think this is Windows (if you have problem with "REQUESTS" & "LXML"). Try to set log_mod = 0 on string 37. 

I test it on Windows 8 and python 3.4.0.

**Comment by alvvvvvin** [comment]
Hi! I'm on OSX Yosemite 10.10.5.



On Thu, Jan 21, 2016 at 5:01 PM, Pasha Lev <notifications@github.com> wrote:

> Hi! What OS do you use? I think this is Windows (if you have problem with
> "REQUESTS" & "LXML"). Try to set log_mod = 0 on string 37.
>
> I test it on Windows 8 and python 3.4.0.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LevPasha/instabot/issues/1#issuecomment-173488569>.
>



-- 
Alvin Yeung  vs.  アルヴィン ヤン

http://al.vvvvv.in/
1-646-675-5792

**Comment by alvvvvvin** [comment]
Posting a GIF of the error or stall that I'm getting when trying to Run the Module from IDLE. Thanks in advance.

![instabot-ticket](https://cloud.githubusercontent.com/assets/3296514/12502652/0f846c52-c10f-11e5-911f-3dd768a642b8.gif)

**Comment by LevPasha** [comment]
Hi! Thx for gif!

I see you have python 2.7.11. I use 3.4 for this, try it!
And try to download last version (""" Instagram bot v 0.02 """). 
I think it solve problem, if.

**Comment by alvvvvvin** [comment]
Hi! I installed 3.4, but then it went back to the lxml error...(earlier, I installed "REQUESTS" & "LXML" for 2.7)

Do you recommend running the script in IDLE?

**Comment by LevPasha** [comment]
Hi you should instal "REQUESTS" & "LXML" for python 3.4 separately. Like pip3 install lxml.
I run in console.

**Comment by alvvvvvin** [comment]
Hi,

I used easy_install to install lxml 3.4.4 but still not working & getting error in module: 

```
Traceback (most recent call last):
  File "/Volumes/Drive 1/MAIN/WORK/APP/instabot-master 3/instabot.py", line 1, in <module>
    from lxml import html
ImportError: No module named 'lxml'
```

**Comment by LevPasha** [comment]
Hi! Download new version (v 0.03). It will work without lxml.

I think you install lxml 3.4.4 for python 2.7.

**Comment by alvvvvvin** [comment]
Hey, thanks for the update. I followed the above but now back w/ the original error:

====== RESTART: /Users/e032888/Downloads/instabot.py-master/instabot.py ======
>>> 

This is utilizing Python 3.4

**Comment by LevPasha** [comment]
Sorry, i don't now anything about python on Mac... Try to run in terminal (console), not from IDLE.

**Comment by kaxil** [comment]
Is it somehow possible that you can change the code for us, or tell us what to change to run this code on python2.7

**Comment by LevPasha** [comment]
Hi kaxil! I start to use python from version 3.4. I don't work with 2.7. What problem for you to setup python 3.4+? If i have time - i try to write changes, to work with python 2.7, but don't know when.

**Comment by kaxil** [comment]
I and most of the other students like me are using python 2.7 because of Proper Documentation & tutorials available on the internet for the same. If you can change the code for 2.7 too, that it would be great. 

Thanks

**Comment by LevPasha** [comment]
![py27](https://cloud.githubusercontent.com/assets/9821835/12808020/e99ec3e6-cb24-11e5-8d39-cb9b463a3afe.png)
Hi! Just try InstaBot on python 2.6 - all ok. I think you can run it on your 2.7 too. Going to change readmy.

**Comment by kaxil** [comment]
ya it is working now, I guess we can add some more functionalities, I will try to fork this git & try to add more features. Did u face any ban from instagram.

And a more documentation would be helpful.

Like if I want to like all the likers of a particular posts how should i achieve it.

or lets say I want to follow the top 10 users with hashtag "lolumlol" , how should I achieve that.

**Comment by LevPasha** [comment]
Hi. I'm happy for you! 

I run bot for some of my friends. It is run about 2 week 24/7, it is about 15000 likes for one account, and no any ban.

When i have some time - i will make some new functional. Will be glad if you fork.

**Comment by kaxil** [comment]
15,000 Likes wow !! , Can I know, how you achieved that much likes, like you use bot to like based on a particular hashtag ? 

Can we get in-touch on Facebook ?

**Comment by LevPasha** [comment]
Bot set 15000 likes! Not 15000 likes on my account. I think bot add about 150-200 likes and 10-20 follower, at day to me. You can find me on fb (same name like on github).

**Comment by alvvvvvin** [comment]
hey @kaxil are you running this on OS X? I just tried the recent build & I'm still getting this error:

```
Traceback (most recent call last):
  File "/Volumes/Drive 1/Downloads_current/instabot.py-master/instabot.py", line 1, in <module>
    import requests
ImportError: No module named requests

```

**Comment by LevPasha** [comment]
@alvvvvvin you should install module `request`. I update readme, how to do it.

**Comment by alvvvvvin** [comment]
Hey @LevPasha thanks. I installed the request module, but now I'm not getting any confirmation if the app is working or not. I'm trying this on terminal on OSX.

Any advice would be appreciated. Thanks!

Posting GIF:

![instabot-ticket_02](https://cloud.githubusercontent.com/assets/3296514/12894719/8c73dea0-ce66-11e5-829f-2a085bf21bf6.gif)

**Comment by alvvvvvin** [comment]
I've tried placing standard usage lines at the end of instabot.py, but have come up w/ these errors:

```
"InstaBot instance has no attribute 'log_mod'" in <bound method InstaBot.__del__ of <__main__.InstaBot instance at 0x1052219e0>> ignored
Traceback (most recent call last):
  File "instabot.py", line 255, in <module>
    log_mod = 1)
TypeError: __init__() got an unexpected keyword argument 'like_in_day'
```

**Comment by taktikal17** [comment]
Yes sadly I got the same error... 
`Traceback (most recent call last):
  File "instabot.py", line 70, in __del__
    self.logout()
  File "instabot.py", line 117, in logout
    self.write_log(log_string)
  File "instabot.py", line 230, in write_log
    if self.log_mod == 0:
AttributeError: 'InstaBot' object has no attribute 'log_mod'
Traceback (most recent call last):
  File "instabot.py", line 255, in <module>
    log_mod = 1)
TypeError: __init__() got an unexpected keyword argument 'like_in_day'`


Python makes me dreaming sometimes.. :'(

**Comment by alvvvvvin** [comment]
@taktikal17 I figured it out:

`like_in_day`

seems to generate an error.

remove it & try.

**Comment by alvvvvvin** [comment]
Hey guys, also wondering if anyone has figured out support for Japanese tags(or non-Latin as for that matter) ?

`SyntaxError: Non-ASCII character '\xe3' in file instabot_02.py on line 263, but no encoding declared; `

**Comment by LevPasha** [comment]
Hi! I think you new in python. Show me what you write (code) in the end of file `instabot.py`.

Write some Japanese tags (i try to do some-thing with it).

**Comment by LevPasha** [comment]
@alvvvvvin specially for you write b87e7ea60a7fdccc8d0c1c743518f54dbd50c901 exception for your error. Now instead of stop program it will write exception `Your text have unicode problem!`.

![2222](https://cloud.githubusercontent.com/assets/9821835/12915361/aeb5185a-cf3b-11e5-9b04-dbde0da485eb.png)

**Comment by cahyonegoro** [comment]
i have the same problem but when i add 
bot = InstaBot('login', 'password',
               like_in_day=1000,
               more_than_likes=10,
               tag_list = ['cat', 'car', 'dog'],
               max_like_for_one_tag=5,
               log_mod = 0)

it become IndentationError: unexpected indent

**Comment by cahyonegoro** [comment]
![capture](https://cloud.githubusercontent.com/assets/16939079/12915540/9b5a438e-cf5e-11e5-9475-acf1600a5ef1.PNG)

**Comment by alvvvvvin** [comment]
hey @LevPasha I added this line:

`# -*- encoding: utf-8 -*-`

to the top, & it works for Japanese now.

**Comment by LevPasha** [comment]
@alvvvvvin ok, happy for you!

**Comment by alvvvvvin** [comment]
@LevPasha Thanks! 

Have you had a chance to review:

`like_in_day`

this seems to break the code from running properly.

**Comment by LevPasha** [comment]
@alvvvvvin download last version b87e7ea60a7fdccc8d0c1c743518f54dbd50c901 i use it myself and have no problem. Or show code, that you write.

**Comment by kaxil** [comment]
@alvvvvvin Instead of `like_in_day` use `like_per_day`, it will work

**Comment by LevPasha** [comment]
@alvvvvvin and @kaxil thx, fix it in readme: bb5b44e8a8df7f7208da3c56cf37497100c31465
